source/coap_service_api.c

```diff
@@ -142,7 +141,8 @@ static uint8_t coap_tx_function(uint8_t *data_ptr, uint16_t data_len, sn_nsdl_ad
     dest_addr.identifier = address_ptr->port;
     dest_addr.type = ADDRESS_IPV6;
 
-    if( -2 == coap_connection_handler_send_data(this->conn_handler, &dest_addr, data_ptr, data_len, (this->service_options & COAP_SERVICE_OPTIONS_SECURE_BYPASS) == COAP_SERVICE_OPTIONS_SECURE_BYPASS) ){
+    if (-2 == coap_connection_handler_send_data(this->conn_handler, &dest_addr, transaction_ptr->have_source_address ? transaction_ptr->source_address:NULL,
```

Comment:
In transaction, local address is static. This flag is set to tell is there really valid address. This should be removed when virtual socet and your implementation is done